frontend/packages/console-app/src/components/nodes/NodeStatus.tsx

```diff
@@ -1,16 +1,67 @@
 import * as React from 'react';
+import * as _ from 'lodash';
+import ConsumerPopover from '@console/shared/src/components/dashboard/utilization-card/TopConsumerPopover';
 import { Status, SecondaryStatus, getNodeSecondaryStatus } from '@console/shared';
 import { NodeKind } from '@console/internal/module/k8s';
+import { humanizeBinaryBytes, humanizeNumber } from '@console/internal/components/utils';
 import { nodeStatus } from '../../status/node';
+import { PressureQueries, Condition } from '../../queries';
+
+const conditionDescriptionMap = Object.freeze({
+  [Condition.DISK_PRESSURE]: 'available memory is low',
+  [Condition.MEM_PRESSURE]: 'available disk capacity is low',
+  [Condition.PID_PRESSURE]: 'CPU is running a large number of processes',
+});
+
+const humanizeMap = Object.freeze({
+  [Condition.DISK_PRESSURE]: humanizeBinaryBytes,
+  [Condition.MEM_PRESSURE]: humanizeBinaryBytes,
+  [Condition.PID_PRESSURE]: humanizeNumber,
+});
+
+const isMonitoredCondition = (condition: string): boolean =>
+  ['DiskPressure', 'PIDPressure', 'MemoryPressure'].includes(condition);
+
+const getDegradedStates = (node: NodeKind): Condition[] => {
+  return node.status.conditions.reduce(
+    (acc: Condition[], elem) =>
+      elem.status === 'True' && isMonitoredCondition(elem.type)
+        ? [...acc, elem.type as Condition]
+        : acc,
+    [],
+  );
+};
+
+const getPopovers = (items: Condition[], node: string): React.ReactElement[] =>
+  items.map((item) => {
+    return (
+      <React.Fragment key={item}>
+        <ConsumerPopover
+          title={_.startCase(item)}
+          current={_.startCase(item)}
+          consumers={PressureQueries[item](node)}
+          humanize={humanizeMap[item]}
+          description={`This node's ${conditionDescriptionMap[item]}. Performance may be degraded.`}
+        />
+        <br />
```

Comment:
`<br />` vs CSS?

Reply:
So currently we don't have a CSS file for NodeStatus and I felt maybe we can go forward without one for now.

Reply:
I'd prefer CSS as well, even if it's a file with one style.